src/leap/bitmask/backend/settings.py

```diff
@@ -122,18 +122,17 @@ def set_selected_gateway(self, provider, gateway):
         self._settings.set(provider, self.GATEWAY_KEY, gateway)
         self._save()
 
-    def get_uuid(self, username):
+    def get_uuid(self, full_user_id):
         """
         Gets the uuid for a given username.
 
         :param username: the full user identifier in the form user@provider
         :type username: basestring
         """
-        leap_assert("@" in username,
+        leap_assert("@" in full_user_id,
                     "Expected username in the form user@provider")
-        user, provider = username.split('@')
-
-        return self._get_value(provider, username, "")
+        username, provider = full_user_id.split('@')
+        return self._get_value(str(provider), str(full_user_id), "")
```

Comment:
is the typecast needed? I think that the ConfigParser object returns str objects.